### PR TITLE
feat: new methods `addBlock` and `deleteBlock` 

### PR DESCRIPTION
**Pull request by valerierose**
## Overview

Adds new methods `addBlock` and `deleteBlock` to block or unblock accounts.

## Example

```
    const result = await client.publishBlock(signer_uuid, 489795);
    console.log(JSON.stringify({ action: "publishBlock", result }));

    // List the blocks for the user
    const blocks = await client.fetchBlockList({ blockedFid: 489795 });
    console.log(JSON.stringify({ action: "fetchBlockList", state: "beforeDeletion", blocks }));

    // Delete the block
    await client.deleteBlock(signer_uuid, 489795);

    // List the blocks again
    const updatedBlocks = await client.fetchBlockList({ blockedFid: 489795 });
    console.log(JSON.stringify({ action: "fetchBlockList", state: "afterDeletion", blocks: updatedBlocks }));
```

```
$ node index.js
{
  "action": "publishBlock",
  "result": {
    "success": true
  }
}
{
  "action": "fetchBlockList",
  "state": "beforeDeletion",
  "blocks": {
    "blocks": [
      {
        "object": "block",
        "blocker": {
          "object": "user",
          "fid": 854074,
          "username": "parallelogram",
          "display_name": "Paralell O. Gram",
          "pfp_url": "https://imagedelivery.net/BXluQx4ige9GuW0Ia56BHw/1c76c0be-30ed-4d6e-424a-0cc4bd0e1000/rectcrop3",
          "custody_address": "0x70882dd0414251c34bda5e89d4e1693ad55c1f3f",
          "profile": {
            "bio": {
              "text": "I have 4 sides and I love web3. Let's chat."
            },
            "location": {
              "latitude": -54.81,
              "longitude": -68.32,
              "address": {
                "city": "Ushuaia",
                "state": "Tierra del Fuego Province",
                "country": "Argentina",
                "country_code": "ar"
              }
            }
          },
          "follower_count": 16,
          "following_count": 9,
          "verifications": [],
          "verified_addresses": {
            "eth_addresses": [],
            "sol_addresses": []
          },
          "verified_accounts": null,
          "power_badge": false
        },
        "blocked_at": "2024-10-31T00:10:05.358Z"
      }
    ],
    "next": {
      "cursor": null
    }
  }
}
{
  "action": "fetchBlockList",
  "state": "afterDeletion",
  "blocks": {
    "blocks": [],
    "next": {
      "cursor": null
    }
  }
}
```

### CODE REVIEW

File: src/neynar-api/v2/openapi-farcaster/apis/block-api.ts [hunk 1]
```diff
@@ -24,13 +24,56 @@ import { BASE_PATH, COLLECTION_FORMATS, type RequestArgs, BaseAPI, RequiredError
...
 export const BlockApiAxiosParamCreator = function (configuration?: Configuration) {
     return {
+        /**
+         * Adds a block for a given FID.
+         * @summary Block FID
+         * @param {BlockReqBody} blockReqBody 
+         * @param {*} [options] Override http request option.
+         * @throws {RequiredError}
+         */
+        addBlock: async (blockReqBody: BlockReqBody, options: RawAxiosRequestConfig = {}): Promise<RequestArgs> => {
```
Comment (Shreyaschorge) [inline src/neynar-api/v2/openapi-farcaster/apis/block-api.ts:45]: Need wrapper code for this method

File: src/neynar-api/v2/openapi-farcaster/apis/block-api.ts [hunk 2]
```diff
@@ -79,6 +122,45 @@ export const BlockApiAxiosParamCreator = function (configuration?: Configuration
...
+            };
+        },
+        /**
+         * Deletes a block for a given FID.
+         * @summary Unblock FID
+         * @param {BlockReqBody} blockReqBody 
+         * @param {*} [options] Override http request option.
+         * @throws {RequiredError}
+         */
+        deleteBlock: async (blockReqBody: BlockReqBody, options: RawAxiosRequestConfig = {}): Promise<RequestArgs> => {
```
Comment (Shreyaschorge) [inline src/neynar-api/v2/openapi-farcaster/apis/block-api.ts:137]: Need wrapper code for this method

File: src/neynar-api/v2/openapi-farcaster/apis/block-api.ts [hunk 3]
```diff
@@ -24,13 +24,56 @@ import { BASE_PATH, COLLECTION_FORMATS, type RequestArgs, BaseAPI, RequiredError
...
  * BlockApi - axios parameter creator
  * @export
  */
 export const BlockApiAxiosParamCreator = function (configuration?: Configuration) {
     return {
+        /**
+         * Adds a block for a given FID.
+         * @summary Block FID
+         * @param {BlockReqBody} blockReqBody 
+         * @param {*} [options] Override http request option.
```
Comment (Shreyaschorge) [inline src/neynar-api/v2/openapi-farcaster/apis/block-api.ts:42]: Need OAS change here
Please add `externalDocs.url` for all the block related API
`https://docs.neynar.com/reference/publish-block`

File: src/neynar-api/v2/openapi-farcaster/apis/block-api.ts (hunk 1 above)
Comment (Shreyaschorge) [inline src/neynar-api/v2/openapi-farcaster/apis/block-api.ts:45]: Need OAS change here
`operationId` should be kebak-case and match the name of the method.
`publish-block`

File: src/neynar-api/v2/openapi-farcaster/apis/block-api.ts (hunk 3 above)
Comment (Shreyaschorge) [reply src/neynar-api/v2/openapi-farcaster/apis/block-api.ts:42]: Same for delete-block

**Review by Shreyaschorge — CHANGES_REQUESTED** [review]

File: src/neynar-api/v2/openapi-farcaster/apis/block-api.ts (hunk 1 above)
Comment (valerierose) [reply src/neynar-api/v2/openapi-farcaster/apis/block-api.ts:45]: Ooops, it was on my local but not pushed. Sorry about that.

**Review by Shreyaschorge — APPROVED** [review]
LGTM

**Pull request merged**